### DISCUSSION TITLE
Pytorch >=1.12, Python >=3.8 Cuda >=11.6

### DISCUSSION THREAD

**Discussion by johnnynunez** [facebookresearch/detectron2#4858] [Show and tell]
Wheels: https://github.com/facebookresearch/detectron2/pull/4857
Refactor: https://github.com/facebookresearch/detectron2/pull/4740

**Comment by johnnynunez** [comment]
@ppwwyyxx @wat3rBro @thiagocrepaldi

**Reply by thiagocrepaldi** [reply]
Hi @johnnynunez  I see that https://github.com/facebookresearch/detectron2/pull/4740/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R122-R143 is a direct copy of my PR https://github.com/facebookresearch/detectron2/pull/4205/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R155-R177. There are other minor parts that were copied too.

Any reason why #4740 doesn't build on top of #4205 [and #4830] instead of creating another PR that concentrates new and copied changes at once? The new CUDA support has been discussed/worked on with detectron2 friends for some time now.

Also, having duplicate PRs waste review time from detectron2 maintainers, not to mention that duplication removes authorship from colleagues that invested time on the original task. In general, huge PRs are hard to review and even harder to revert if an unforeseen issue arises.

**Reply by johnnynunez** [reply]
> Hi @johnnynunez  I see that https://github.com/facebookresearch/detectron2/pull/4740/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R122-R143 is a direct copy of my PR https://github.com/facebookresearch/detectron2/pull/4205/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R155-R177. There are other minor parts that were copied too.
> 
> 
> 
> Any reason why #4740 doesn't build on top of #4205 [and #4830] instead of creating another PR that concentrates new and copied changes at once? The new CUDA support has been discussed/worked on with detectron2 friends for some time now.
> 
> 
> 
> Also, having duplicate PRs waste review time from detectron2 maintainers, not to mention that duplication removes authorship from colleagues that invested time on the original task. In general, huge PRs are hard to review and even harder to revert if an unforeseen issue arises.

I first tried your idea but it didn't work for me. After this, I created a ticket for CircleCi because cuda images not updated since 2021, so now there images with cuda 11.8 and 11.6 at the same image and they are working for cuda 12.0. When I was using this new image, I had problem for one test with pytorch +1.10, so I fixed it with skip. 
So, I finally went my way and made the appropriate changes and updates for use. The problem is that it was uploaded with the code changed since I updated all the libraries to the latest version, including black, isort, and flake8 and python 3.11

I am using detectron2 with my version on a daily basis because otherwise it forced me to install older versions of pytorch and cuda in my master course

**Comment by johnnynunez** [comment]
@ppwwyyxx @wat3rBro @thiagocrepaldi 
I have deleted the previous two and redone it to make the changes clearer.
https://github.com/facebookresearch/detectron2/pull/4868

**Comment by MiroPsota** [comment]
As an alternative, you can try my repository for building packages and PyPI simple index and see if it works for you:
https://github.com/facebookresearch/detectron2/discussions/5200
